codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/protocols/HttpBoundProtocolGenerator.kt

```diff
@@ -399,199 +378,3 @@ class HttpBoundProtocolTraitImplGenerator(
-            "marshallerConstructorFn" to marshallerConstructorFn,
-            "OperationError" to operationShape.errorSymbol(symbolProvider)
-        )
-    }
-
-    private fun RustWriter.serializeViaPayload(
-        bodyMetadata: BodyMetadata,
-        member: MemberShape,
-        serializerGenerator: StructuredDataSerializerGenerator
-    ) {
-        val fnName = "ser_payload_${member.container.name.toSnakeCase()}"
-        val ref = when (bodyMetadata.takesOwnership) {
-            true -> ""
-            false -> "&"
-        }
-        val serializer = RuntimeType.forInlineFun(fnName, operationSerModule) {
-            it.rustBlockTemplate(
-                "pub fn $fnName(payload: $ref #{Member}) -> std::result::Result<#{SdkBody}, #{BuildError}>",
-                "Member" to symbolProvider.toSymbol(member),
-                *codegenScope
-            ) {
-                // If this targets a member & the member is None, return an empty vec
-                val ref = when (bodyMetadata.takesOwnership) {
-                    false -> ".as_ref()"
-                    true -> ""
-                }
-
-                if (symbolProvider.toSymbol(member).isOptional()) {
-                    withBlockTemplate(
-                        """
-                        let payload = match payload$ref {
-                            Some(t) => t,
-                            None => return Ok(#{SdkBody}::from(""",
-                        "))};",
-                        *codegenScope
-                    ) {
-                        when (val targetShape = model.expectShape(member.target)) {
-                            is StringShape, is BlobShape, is DocumentShape -> rust("".dq())
-                            is StructureShape -> rust("#T()", serializerGenerator.unsetStructure(targetShape))
-                        }
-                    }
-                }
-                // When the body is a streaming blob it _literally_ is a SdkBody already
-                // mute this clippy warning to make the codegen a little simpler
-                Attribute.Custom("allow(clippy::useless_conversion)").render(this)
-                withBlock("Ok(#T::from(", "))", RuntimeType.sdkBody(runtimeConfig)) {
-                    renderPayload(member, "payload", serializerGenerator)
-                }
-            }
-        }
-        rust("#T($ref self.${symbolProvider.toMemberName(member)})?", serializer)
-    }
-
-    private fun RustWriter.renderPayload(
-        member: MemberShape,
-        payloadName: String,
-        serializer: StructuredDataSerializerGenerator
-    ) {
-        when (val targetShape = model.expectShape(member.target)) {
-            // Write the raw string to the payload
-            is StringShape -> {
-                if (targetShape.hasTrait<EnumTrait>()) {
-                    rust("$payloadName.as_str()")
-                } else {
-                    rust("""$payloadName.to_string()""")
-                }
```

Comment:
I am not sure I understand the implication of changing the return type of the payload from a str / String to a vec.. (see same function in HttpBoundProtocolPayloadGenerator)

Reply:
I added the explanation to the PR description and the commit message of https://github.com/awslabs/smithy-rs/pull/1158/commits/0861b19eaf2394026dcff60a05ff3aeb9d1f2631, which is the one I'll use when I squash the whole patchset.